### PR TITLE
Dotnet runner + DiffSharp module 

### PR DESCRIPTION
**Pull request by novikov-alexander**
- Dotnet runner similar to C++ one
- Dotnet runner tests
- Dotnet runner integration
- DiffSharp module supporting BA problem
- Tests for DiffSharp module

### CODE REVIEW

**Comment by novikov-alexander** [comment]
I have to add plots and fix wrong merge in h and cpp files

File: ADBench/run-all.ps1 [hunk 1]
```diff
@@ -446,13 +490,13 @@ $tool_descriptors = @(
     #[Tool]::new("CeresEigen", "bin", [ObjectiveType] "Hand", $false, 0.0, $false, $true)
     [Tool]::new("Finite", "cpp", [ObjectiveType] "GMM, BA, Hand, LSTM", $true, 1e-4)
     [Tool]::new("FiniteEigen", "cpp", [ObjectiveType] "Hand", $true, 1e-4)
-    [Tool]::new("Manual", "cpp", [ObjectiveType] "GMM, BA, Hand, LSTM", $false, 0.0)
-    [Tool]::new("ManualEigen", "cpp", [ObjectiveType] "GMM, BA, Hand, LSTM", $true, $default_tolerance)
-    [Tool]::new("ManualEigenVector", "cpp", [ObjectiveType] "GMM", $true, $default_tolerance)
+    #[Tool]::new("Manual", "cpp", [ObjectiveType] "GMM, BA, Hand, LSTM", $false, 0.0)
```
Comment (mikhailnikolaev) [inline ADBench/run-all.ps1]: Why these tools are commented?

File: src/cpp/modules/manualEigen/gmm.h [hunk 2]
```diff
@@ -0,0 +1,273 @@
+#pragma once
```
Comment (mikhailnikolaev) [inline src/cpp/modules/manualEigen/gmm.h]: And why such files are changed in this PR?

File: src/dotnet/modules/DiffSharpModule/build.bat [hunk 3]
```diff
@@ -0,0 +1,7 @@
+set srcdir=%1
```
Comment (mikhailnikolaev) [inline src/dotnet/modules/DiffSharpModule/build.bat]: Why is this file needed?

File: src/dotnet/utils/JacobianComparisonLib/JacobianComparison.cs [hunk 4]
```diff
@@ -181,6 +181,23 @@ public string ToJsonString()
 }}";
         }
 
+        public string ToJsonString()
```
Comment (mikhailnikolaev) [inline src/dotnet/utils/JacobianComparisonLib/JacobianComparison.cs]: Code duplication

File: tools/Manual/gmm_d-.cpp [hunk 5]
```diff
@@ -0,0 +1,437 @@
+#include "gmm_d.h"
```
Comment (mikhailnikolaev) [inline tools/Manual/gmm_d-.cpp]: As I understand this file is not in a master anymore

File: tools/Manual/gmm_d-.h [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#pragma once
```
Comment (mikhailnikolaev) [inline tools/Manual/gmm_d-.h]: As above

**Comment by mikhailnikolaev** [comment]
As I understand there were some problems with rebase or something like this.
